runtime-modules/storage/src/lib.rs

```diff
@@ -4404,36 +4394,55 @@ impl<T: Config> Module<T> {
             .and_then(|x| x)
     }
 
-    fn construct_objects_to_remove(
+    fn prepare_data_objects_to_insert(
+        objects: Vec<DataObjectOf<T>>,
+        bloat_bonds: Vec<RepayableBloatBondOf<T>>,
+    ) -> Result<Vec<DataObjectOf<T>>, DispatchError> {
+        let mut i = 0u64;
+        objects
+            .iter()
+            .map(|o| {
+                let state_bloat_bond = bloat_bonds
+                    .get(i as usize)
```

Comment:
Same issue here with the uncoordinate bloat bond and object lists,

Reply:
Fixed in https://github.com/Joystream/joystream/pull/4164/commits/deaf7f35d2f97f1068c51df5364962d755b043b2